### ISSUE TITLE
performance issues

### ISSUE THREAD

**Issue by immysl** [immysl/ng2use#1]
I noticed a few issues related to performance. I have listed them below:
- The initial app loading time: this could be because we aren't using server-rendering and also the Angular CLI build is not optimized enough. So maybe we could move to a webpack-based build now that Angular 2 has described on how to use in their official docs.
- The rendering: this is probably due to the Materlialize CSS library that we are using. We'll need to replace it with something more performant.
- The loading of data when a module is expanded: we'll need to find a way to optimize this further. Currently we are caching the service call so it is fast when we expand it for a second time. However, the first time load has more room for improvement.

So these are the issues that I've found out for now. If you come across anymore please list them down so that we can look into them one by one and make them site much faster than what it is now. :smiley:

**Comment by immysl** [comment]
I have made some progress on this. The build is now based on webpack instead of Angular CLI. Last I checked, Angular CLI has made some good progress but they haven't been able to get tree shaking to work yet. So let's stick with webpack for now :smile: 

Also, I have currently removed jQuery and also in the process of removing material icons and replacing it with a custom made font to reduce the size dramatically.

And I didn't remove Materialize library and instead have integrated tree shaking for CSS with webpack. Unfortunately, it isn't removing all the unused CSS. So I will have to look into this further and figure out how to make tree shaking work much better than that.

And one more thing. You'll notice the accordion isn't working properly. That is because Materialize needs jQuery for it to work. But I have removed Materialize JS as well :stuck_out_tongue: I will try to implement a simple accordion with Angular 2.
